File: project/ArtExbitionFrontend/ArtExbition/src/app/models/gallery.ts
import { Artist } from "./artist";
import { Artwork } from "./artwork";

export class Gallery {

    name: string | undefined;
    description?: string;
    location?: string;
    artistId: number | undefined; // Foreign Key
    artist?: Artist; // Navigation Property
    artworkGalleries?: Artwork[]; // List of artworks in the gallery
}

File: ArtExhibitionPlatform/ArtExhibitionPlatform.Domain/Gallery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ArtexibitionPlatform.Domain
{
    public class Gallery
    {
        public int GalleryID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }

        // Foreign Key
        public int ArtistID { get; set; }
        public Artist Artist { get; set; }

        // Navigation Properties
        public ICollection<ArtworkGallery> ArtworkGalleries { get; set; } = new List<ArtworkGallery>();
    }
}

